AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/UsersRoutes.java

```diff
@@ -103,6 +103,32 @@ public UserApiModel updateEntity(Request req, Response res, int id) {
         return mapModelToApiModel(user);
     }
 
+    public UserApiModel updateUsernamePasswordEntity(Request req, Response res, int id) {
+        UserApiModel userApiModel = JsonRequestUtils.readBodyAsType(req, UserApiModel.class);
+
+        permissionsVerifier.verifyIsAdminOrRelevantUser(req, id);
+
+        Optional<User> optionalUser = usersDao.getUser(id);
+        if (!optionalUser.isPresent()) {
+            throw new FailedRequestException(ErrorCode.INVALID_INPUT, "userId cannot be found");
+        }
+
+        User oldUser = optionalUser.get();
+
+        User user = new User(
```

Comment:
Would this be slightly clearer without having to create a new User?

E.g. we could just directly update the old user:

``` 
User currentUser = optionalUser.get();
if(Strings.isNullOrEmpty(userApiModel.getPassword()){
  currentUser.setHashedPassword( //hash ... );
}

currentUser.setUsername(userApiModel.getUsername());

usersDao.updateUser(currentUser);
```

Reply:
not too sure how to implement this and if its necessary as we based it off of updateEntity.